lib/opte/src/engine/packet.rs

```diff
@@ -111,52 +112,102 @@ pub static FLOW_ID_DEFAULT: InnerFlowId = InnerFlowId {
     PartialOrd,
     Serialize,
 )]
+#[repr(C)]
 pub struct InnerFlowId {
-    pub proto: Protocol,
-    pub src_ip: IpAddr,
+    // Using a `u8` here for `proto` hides the enum repr from SDTs,
+    // but we need to have a `u16` discriminant for the IPs to set up
+    // 4B alignment on inaddr6_t on the dtrace side (so we lose out
+    // on the possibility of 38B packing).
+    pub proto: u8,
+    pub addrs: AddrPair,
```

Comment:
`AddrPair`'s discriminant is a `u16` right now, so there's 1B of padding between `proto` and `addrs` here. Feels like this comment is out of date?

Also, does the alignment requirement on the dtrace side matter since we copy to a local alloca anyways in `FLOW_FMT`?

Reply:
The comment is written very confusingly, my bad. Should be fixed now.

The alignment does unfortunately matter, unless we just use byte arrays for the actual v4/v6 addresses in `common.d`. Alignment is set [here](https://github.com/oxidecomputer/illumos-gate/blob/stlouis/usr/src/uts/common/netinet/in.h#L127). In `common.d` we have:
```
typedef struct flow_id_sdt_arg {
	uint8_t		proto;
	uint16_t	af;
	union addrs {
		struct {
			ipaddr_t src;
			ipaddr_t dst;
		} ip4;
		struct {
			in6_addr_t src;
			in6_addr_t dst;
		} ip6;
	} addrs;
	uint16_t	src_port;
	uint16_t	dst_port;
} flow_id_sdt_arg_t;
```
So `addrs` will be 4B-aligned due to `ip6`, and we get a bad read on the initial alloca.